src/main/kotlin/com/exactpro/th2/check1/rule/nomessage/NoMessageCheckTask.kt

```diff
@@ -43,41 +43,40 @@ class NoMessageCheckTask(
     eventBatchRouter: MessageRouter<EventBatch>
 ) : AbstractCheckTask(ruleConfiguration, startTime, sessionKey, parentEventID, messageStream, eventBatchRouter) {
 
-    private val protoPreMessageFilter: RootMessageFilter = protoPreFilter.toRootMessageFilter()
-    private val messagePreFilter = SailfishFilter(
-        CONVERTER.fromProtoPreFilter(protoPreMessageFilter),
-        protoPreMessageFilter.toCompareSettings()
+    private var protoPreMessageFilter: RootMessageFilter? = protoPreFilter.toRootMessageFilter()
+    private var messagePreFilter: SailfishFilter? = SailfishFilter(
+        CONVERTER.fromProtoPreFilter(protoPreMessageFilter!!),
+        protoPreMessageFilter!!.toCompareSettings()
     )
 
-    private val metadataPreFilter: SailfishFilter? = protoPreMessageFilter.metadataFilterOrNull()?.let {
+    private var metadataPreFilter: SailfishFilter? = protoPreMessageFilter!!.metadataFilterOrNull()?.let {
         SailfishFilter(
             CONVERTER.fromMetadataFilter(it, VerificationUtil.METADATA_MESSAGE_NAME),
             it.toComparisonSettings()
         )
     }
 
-    private lateinit var preFilterEvent: Event
-    private lateinit var resultEvent: Event
+    private var preFilterEvent: Event? = null
+    private var resultEvent: Event? = null
 
     private var extraMessagesCounter: Int = 0
 
-
     override fun onStart() {
         super.onStart()
         preFilterEvent = Event.start()
             .type("preFiltering")
-            .bodyData(protoPreMessageFilter.toTreeTable())
+            .bodyData(protoPreMessageFilter!!.toTreeTable())
         rootEvent.addSubEvent(preFilterEvent)
         resultEvent = Event.start()
             .type("noMessagesCheckResult")
         rootEvent.addSubEvent(resultEvent)
     }
 
     override fun Observable<MessageContainer>.taskPipeline(): Observable<MessageContainer> =
-        preFilterBy(this, protoPreMessageFilter, messagePreFilter, metadataPreFilter, LOGGER) { preFilterContainer -> // Update pre-filter state
+        preFilterBy(this, protoPreMessageFilter!!, messagePreFilter!!, metadataPreFilter, LOGGER) { preFilterContainer -> // Update pre-filter state
```

Comment:
Please, do not use `!!` in code. That indicates that something is wrong with the code. Also, it does not report any useful messages in case of error. It is better to use `checkNotNull` for that